zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParser.scala

```diff
@@ -211,27 +243,471 @@ object JSONParser {
 
 
   // TODO find better name for 'global'
-  implicit val queryFormat: Format[Query] = (
+  implicit val queryFormat: Format[UnresolvedQuery] = (
     (JsPath \ "dataset").format[String] and
-      (JsPath \ "lookup").formatNullable[Seq[LookupStatement]].inmap[Seq[LookupStatement]](
-        o => o.getOrElse(Seq.empty[LookupStatement]),
+      (JsPath \ "lookup").formatNullable[Seq[UnresolvedLookupStatement]].inmap[Seq[UnresolvedLookupStatement]](
+        o => o.getOrElse(Seq.empty[UnresolvedLookupStatement]),
         s => if (s.isEmpty) None else Some(s)
       ) and
-      (JsPath \ "filter").formatNullable[Seq[FilterStatement]].inmap[Seq[FilterStatement]](
-        o => o.getOrElse(Seq.empty[FilterStatement]),
+      (JsPath \ "filter").formatNullable[Seq[UnresolvedFilterStatement]].inmap[Seq[UnresolvedFilterStatement]](
+        o => o.getOrElse(Seq.empty[UnresolvedFilterStatement]),
         s => if (s.isEmpty) None else Some(s)
       ) and
-      (JsPath \ "unnest").formatNullable[Seq[UnnestStatement]].inmap[Seq[UnnestStatement]](
-        o => o.getOrElse(Seq.empty[UnnestStatement]),
+      (JsPath \ "unnest").formatNullable[Seq[UnresolvedUnnestStatement]].inmap[Seq[UnresolvedUnnestStatement]](
+        o => o.getOrElse(Seq.empty[UnresolvedUnnestStatement]),
         s => if (s.isEmpty) None else Some(s)
       ) and
-      (JsPath \ "group").formatNullable[GroupStatement] and
-      (JsPath \ "select").formatNullable[SelectStatement] and
-      (JsPath \ "global").formatNullable[GlobalAggregateStatement] and
+      (JsPath \ "group").formatNullable[UnresolvedGroupStatement] and
+      (JsPath \ "select").formatNullable[UnresolvedSelectStatement] and
+      (JsPath \ "global").formatNullable[UnresolvedGlobalAggregateStatement] and
       (JsPath \ "estimable").formatNullable[Boolean].inmap[Boolean](
         o => o.getOrElse(false),
         s => if (s) Some(s) else None
       )
-    ) (Query.apply, unlift(Query.unapply))
+    ) (UnresolvedQuery.apply, unlift(UnresolvedQuery.unapply))
+
+
+  implicit def toUnresolved(query: Query): UnresolvedQuery =
```

Comment:
It's a use of scala implicit conversion. This function will be treated as a type convertor automatically by scala.